pkg/reconciler/testing/mongodbsource.go

```diff
@@ -75,6 +76,20 @@ func WithMongoDbSourceSink(uri *apis.URL) MongoDbSourceOption {
 	}
 }
 
+// WithMongoDbSourceConnectionFailed updates the status of the connection to be failed.
+func WithMongoDbSourceConnectionFailed(err string) MongoDbSourceOption {
+	return func(s *v1alpha1.MongoDbSource) {
+		s.Status.MarkConnectionFailed(errors.New(err))
+	}
+}
+
+// WithMongoDbSourceConnectionSuccess updates the status of the connection to be successful.
+func WithMongoDbSourceConnectionSuccess(err string) MongoDbSourceOption {
```

Comment:
This shoudnt use err string as argument, will delete in next UT PR

Reply:
ok